File: c#/classes/ClassStaticFunctionAccess.cs
class Token {
  public static int Number;
  public string Color;

  public void Update(int number, string color) {
    Number = number;
    Color = color;
  }

  public static void Reset() {
    Number = 0;
  }
}

class ClassStaticFunctionAccess {
  public static void Main(string[] args) {
    Token.Reset();
    System.Console.WriteLine("Token.Number: " + Token.Number);

    System.Console.WriteLine();

    Token some_token = new Token();

    some_token.Update(5, "magenta");
    System.Console.WriteLine("some_token.Color: " + some_token.Color);
    System.Console.WriteLine("Token.Number: " + Token.Number);

    System.Console.WriteLine();

    ref Token reference_to_token = ref some_token;

    reference_to_token.Update(6, "amaranthine");
    System.Console.WriteLine("some_token.Color: " + some_token.Color);
    System.Console.WriteLine("Token.Number: " + Token.Number);

    System.Console.WriteLine();
  }
}

File: php/classes/class_static_function_access.php
<?php
/*
 * Copyright (C) 2019-2026 Katarzyna Krasińska
 * Languagium - https://github.com/katheroine/languagium
 * Licensed under GPL-3.0 - see LICENSE.md
 */

class Token {
    public static int $number;
    public string $color = "";
    public function update(int $number, string $color): void {
        self::$number = $number;
        $this->color = $color;
    }
    public static function reset(): void {
        self::$number = 0;
    }
}

Token::reset();
print("Token::number: " . Token::$number . "\n");

print("\n");

$some_token = new Token();

$some_token->update(5, "magenta");
print("\$some_token->color: {$some_token->color}\n");
print("Token::number: " . Token::$number . "\n");

print("\n");

$reference_to_token = &$some_token;

$reference_to_token->update(6, "amaranthine");
print("\$some_token->color: {$some_token->color}\n");
print("Token::number: " . Token::$number . "\n");

print("\n");
